### PR TITLE
ci: Rng tests refactor

### PR DESCRIPTION
**Pull request by MikhailKardash**
## Description

RNG restore tests can be done without master since that is all done by the trial controller when a checkpoint is loaded.

## Test Plan

Check that `test_unit` is all green in CircleCI. 

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
MLG-570

### CODE REVIEW

File: harness/tests/experiment/fixtures/estimator_no_op/model_def.py [hunk 1]
```diff
@@ -27,7 +26,6 @@ def __init__(self, ctx):
 
     def on_trial_close(self) -> None:
         if self.ctx.distributed.get_rank() == 0:
-            time.sleep(15)
```
Comment (rb-determined-ai) [inline harness/tests/experiment/fixtures/estimator_no_op/model_def.py:30]: I don't mind dropping this sleep, but it was for a specific regression.  If we feel it's no longer necessary, then probably there's a test built around this that is also unnecessary.

The regression was that we were killing all workers as soon as any worker said it was done.  (see commit `1c8328d07e482345e6d83325d2e9b404207ee873`).

But that was several refactors ago, this _really_ shouldn't be a risk anymore.  Our termination sequence doesn't resemble the old one.

So yes, please also remove `test_on_trial_close_callback`.

This sleep was intentional actually.

Gotta remember how it was failing before... not sure if it's still a regression test worth having.

File: harness/tests/experiment/keras/test_tf_keras_trial.py [hunk 2]
```diff
@@ -509,3 +514,75 @@ def make_workloads() -> workload.Stream:
     )
     with pytest.raises(ValueError, match="incompatible"):
         controller.run()
+
+
+@pytest.mark.tensorflow
+@pytest.mark.tf1_cpu
+def test_rng_restore(tmp_path: Path):
```
Comment (rb-determined-ai) [inline harness/tests/experiment/keras/test_tf_keras_trial.py:521]: if you are going to run multiple RNG tests inside the same process, can you set the global RNGs _wrong_ in between TrialControllers, so it's impossible to get a false-positive test?

File: harness/tests/experiment/fixtures/estimator_no_op/model_def.py (hunk 1 above)
Comment (MikhailKardash) [reply harness/tests/experiment/fixtures/estimator_no_op/model_def.py:30]: Removed. Didn't know about the workers thing, but the sleep was just making this test run for ~45 seconds, which seems excessive.

File: harness/tests/experiment/keras/test_tf_keras_trial.py [hunk 3]
```diff
@@ -509,3 +515,89 @@ def make_workloads() -> workload.Stream:
...
+    )
+
+    trial_A_controller.run()
+
+    # reset random seed after checkpointing
+    other_random_seed = random.randint(0, 1 << 31)
+    while other_random_seed == 325:
+        other_random_seed = random.randint(0, 1 << 31)
+
+    trial_A_controller._set_random_seeds(other_random_seed)
```
Comment (rb-determined-ai) [inline harness/tests/experiment/keras/test_tf_keras_trial.py:571]: why not simplify all this to just one:
```python
trial_A_controller._set_random_seeds(0)
```
?

I only care that the RNG state  can't accidentally pick up from where it left off.

**Review by rb-determined-ai — APPROVED** [review]

File: harness/tests/experiment/keras/test_tf_keras_trial.py (hunk 3 above)
Comment (MikhailKardash) [reply harness/tests/experiment/keras/test_tf_keras_trial.py]: Yeah that's better. Updated all tests for this.

**Pull request merged**